### DISCUSSION TITLE
Issue with visualizing using TensorBoard

### DISCUSSION THREAD

**Discussion by StevenvdKamp** [tudelft/AE4353-Y24#27] [Ex_2]
Hello,

My issue: TensorBoard sessions returns a blank page.

I am trying to visualize the training metrics that are obtained in the second to last code block of the notebook. As far as I understand it these metrics are saved in events files in the folder `ex_2/runs/`.
When the code block is finished with running, vscode automatically prompts to launch a TensorBoard session. It asks you to select the location of the events files and then opens up a port that should be able to be opened in the web browser (http://localhost:6006/). Unfortunately for me this only returns a totally blank page. (Note: It does _not_ show that the site can't be reached)

Furthermore, doing it by hand with the command `tensorboard --logdir=ex_2/runs/test/` in the bash terminal in vscode gives me the same results; A fully blank page.
It does look like that I have specified the correct logdir, as when I run the command `tensorboard --logdir=ex_2/runs/test/ --inspect` it returns something in the form of:

> Found event files in:
> ex_2/runs/test/
> 
> These tags are in ex_2/runs/test/:
> audio - AudioSummary
> histograms - HistorgramSummary
> images - ImageSummary
> scalars - ScalarSummary

Could somebody help me with get the TensorBoard visualizations working? Is there something I am missing or am doing wrong? Should I switch to a different visualization technique? 

Thank you very much in advance!

**Comment by yilun-wu** [comment]
Hi,
It is weird that it returns a blank page. Typically, when you click the "Launch Tensorboard Session" in vscode, it will open a new tab showing you the tensorboard UI. Even if you select the wrong log dir (which you did correctly), it will still show the UI.
My guess is that your installation of tensorboard might not be complete. Perhaps try to remove the tensorboard package and install it again.
Otherwise, you may switch to another logging tool: Weights and Biases. See the kaggle version of ex_2 notebook for some guide on that.
